backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/impl/LuceneBackendImplementor.java

```diff
@@ -1,22 +0,0 @@
-/*
- * Hibernate Search, full-text search for your domain model
- *
- * License: GNU Lesser General Public License (LGPL), version 2.1 or later
- * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
- */
-package org.hibernate.search.v6poc.backend.lucene.impl;
-
-import org.hibernate.search.v6poc.backend.lucene.document.impl.LuceneRootDocumentBuilder;
-import org.hibernate.search.v6poc.backend.lucene.index.impl.IndexingBackendContext;
-import org.hibernate.search.v6poc.backend.lucene.search.query.impl.SearchBackendContext;
-import org.hibernate.search.v6poc.backend.spi.BackendImplementor;
-
-/**
- * @author Guillaume Smet
- */
-public interface LuceneBackendImplementor extends BackendImplementor<LuceneRootDocumentBuilder> {
```

Comment:
Not really fond of removing this interface. I like to have a common interface for all the different backend implementations of Lucene as we will probably have more than one.

Reply:
The backend is at the root of everything, so it depends (transitively) on almost everything. Thus, if we reference the backend from anywhere else, we will have a cyclic dependency.
Therefore we should not reference the backend from anywhere, and having an interface for backends is useless.

I can add an abstract class if you want, but even if we end up with configurable implementation bits for Lucene, I would rather move the specific bits to a separate object and use composition instead of multiple backend implementations.
Like you did for the multitenancy strategy, and like I did with the DirectoryProvider in this PR, for instance. We won't have one backend type per Directory type, we will just have a directory provider of a specific type inside the backend. If we need to make another, orthogonal concept configurable, we can then introduce another object inside the backend, whereas if we were handling these concepts in the backend directly, we would end up with a combinatorial explosion in the number of backend implementations.

Reply:
OK, sounds good.